# HG changeset patch
# User Kai Habel
# Date 1233345152 18000
#      Fri Jan 30 14:52:32 2009 -0500
# Node ID 579f1f54cc3a871518cd18bd095c4827d0f4b46a
# Parent  fac8c78b4fb9231c748d0bd6b9d14574d37e6524
Add support for transparent surfaces

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,14 @@
+2009-01-30  Kai Habel  <kai.habel@gmx.de>
+
+	* plot/__go_draw_axes__.m: Add support for transparent surfaces.
+	* plot/__gnuplot_has_feature__.m: Add feature 'transparent_surface',
+	Require gnuplot 4.3 for transparent patches and surfaces.
+
 2009-01-30  Benjamin Lindner  <lindnerben@gmx.net>
 
 	* time/datestr.m: Convert YYYY to %Y instead of %C%y.
 	Convert [Dd][Dd] to %d instead of %e.
 
 2009-01-30  Ben Abbott  <bpabbott@mac.com>
 
 	* plot/print.m: Use __gnuplot_has_feature__ instead of
diff --git a/scripts/plot/__gnuplot_has_feature__.m b/scripts/plot/__gnuplot_has_feature__.m
--- a/scripts/plot/__gnuplot_has_feature__.m
+++ b/scripts/plot/__gnuplot_has_feature__.m
@@ -21,23 +21,24 @@
 
 ## Author: Ben Abbott <bpabbott@mac.com>
 ## Created: 2009-01-27
 
 function res = __gnuplot_has_feature__ (feature)
   persistent features has_features
   features = {"x11_figure_position",
               "transparent_patches",
+              "transparent_surface",
               "epslatex_implies_eps_filesuffix",
               "epslatexstandalone_terminal"};
 
   if (isempty (has_features))
     gnuplot_version = __gnuplot_version__ ();
-    versions = {"4.2.4", "4.2.4", "4.2", "4.2"};
-    operators = {">", ">", ">=", ">="};
+    versions = {"4.2.4", "4.3", "4.3", "4.2", "4.2"};
+    operators = {">", ">=", ">=", ">=", ">="};
     have_features = logical (zeros (size (features)));
     for n = 1 : numel (have_features)
       has_features(n) = compare_versions (gnuplot_version, versions{n}, operators{n});
     endfor
   endif
 
   n = find (strcmpi (feature, features));
   if (isempty (n))
diff --git a/scripts/plot/__go_draw_axes__.m b/scripts/plot/__go_draw_axes__.m
--- a/scripts/plot/__go_draw_axes__.m
+++ b/scripts/plot/__go_draw_axes__.m
@@ -870,16 +870,22 @@ function __go_draw_axes__ (h, plot_strea
 	      dord = "depthorder";
 	    endif
 
 	    if (flat_interp_face && strncmp (obj.edgecolor, "flat", 4))
               fprintf (plot_stream, "set pm3d explicit at s %s %s corners2color c3;\n", 
 		       interp_str, dord);
             elseif (!facecolor_none_or_white)
               if (strncmp (obj.edgecolor, "none", 4))
+                if (__gnuplot_has_feature__ ("transparent_surface") 
+                    && isscalar (obj.facealpha))
+                  fprintf (plot_stream,
+                           "set style fill transparent solid %f;\n",
+                           obj.facealpha);
+                endif
                 fprintf (plot_stream, "set pm3d explicit at s %s corners2color c3;\n", 
 			 interp_str, dord);
               else
                 edgecol = obj.edgecolor;
                 if (ischar (obj.edgecolor))
                   edgecol = [0, 0, 0];
                 endif
                 fprintf (plot_stream, "set pm3d explicit at s hidden3d %d %s %s corners2color c3;\n", 
@@ -889,16 +895,22 @@ function __go_draw_axes__ (h, plot_strea
 		  colorspec = "";
 		else
 		  colorspec = sprintf ("linecolor rgb \"#%02x%02x%02x\"",
 				       round (255*edgecol));
 		endif
                 fprintf (plot_stream,
                          "set style line %d %s lw %f;\n",
                          data_idx, colorspec, obj.linewidth);
+                if (__gnuplot_has_feature__ ("transparent_surface") 
+                    && isscalar (obj.facealpha))
+                  fprintf (plot_stream,
+                           "set style fill transparent solid %f;\n",
+                           obj.facealpha);
+                endif
               endif
             endif
 	  endif
 
 	case "text"
 	  [label, f, s] = __maybe_munge_text__ (enhanced, obj, "string");
 	  if (strcmp (f, "*"))
 	    fontspec = "";
